### DISCUSSION TITLE
Is there an equivalent access to GenerateSignalUrl() from an activity in Studio or do we need to create a custom activity?

### DISCUSSION THREAD

**Discussion by 7studios** [elsa-workflows/elsa-studio#228] [General]
Is there an equivalent access to GenerateSignalUrl() from an activity in Studio or do we need to create a custom activity?

Thank you,

Greg

**Comment by glime-ai** [comment]
In Elsa 3, the equivalent of `GenerateSignalUrl()` from Elsa 2 is achieved through the `generateEventTriggerUrl` JavaScript function, which can be used within Elsa Studio. This function allows you to generate a URL that triggers a given event name scoped to the current workflow instance. You do not need to create a custom activity for this purpose in Elsa Studio.

If you need to generate such URLs programmatically within a custom activity, you would need to implement the logic to construct the URL and potentially secure it with a SAS token or similar mechanism. The details of this implementation would depend on your specific requirements and the context of your workflow.

For more information on how to use `generateEventTriggerUrl`, you can refer to the GitHub issue where it is mentioned ([#4475](https://github.com/elsa-workflows/elsa-core/issues/4475)).

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
